### PR TITLE
Vmo 3370 make connection activable

### PR DESCRIPTION
**Pull request by safydy-r**
Dev notes:
- the Learderline library doesn't provide a native mouse event handler on lines, I had to create from scratch

### CODE REVIEW

File: src/components/interaction-designer/Connection.vue [hunk 1]
```diff
@@ -7,7 +7,19 @@
   // import LeaderLine from 'leader-line'
   const {LeaderLine} = window
   import {set} from 'lodash'
-  import {mapGetters} from 'vuex'
+import {mapState, mapGetters, mapMutations} from 'vuex'
+
+const categoryColorMappings = {
```
Comment (safydy-r) [inline src/components/interaction-designer/Connection.vue:12]: mouved out from `mounted()`

File: src/components/interaction-designer/Connection.vue [hunk 2]
```diff
@@ -24,12 +36,57 @@
...
         // line: null, // no need to set up observers over this
+        isPermanentlyActive: false,
       }
     },
 
     computed: {
+      ...mapState('builder', ['activeConnectionContext']),
       ...mapGetters('builder', ['blocksById']),
 
+      options() {
```
Comment (safydy-r) [inline src/components/interaction-designer/Connection.vue:47]: mouved out from mounted()

**Review by bzabos — APPROVED** [review]
This is awesome! I love that you provided the ability to handle multiple connections activated concurrently. This essentially paves architecture to handle multiple blocks activated as well if we leave `connectionContexts`' properties nullable and use this instead of `activatedBlockId`.

File: src/components/interaction-designer/Connection.vue (hunk 2 above)
Comment (safydy-r) [reply src/components/interaction-designer/Connection.vue:47]: @bzabos , I see what you meant. 
I made 1h research but cannot find similar `init()` possibility. For now, I would stick to current implementation. But if you find it very important we can create a new ticket to handle extra research.

I'll move this to QA for now as I consider this as minor concern.
Let me know your thoughts later when you see the comment.

**Comment by adeablay** [comment]
@safydy I got this console error while trying to relocate a connection to another block.
When a block is already connected to another block, I can only reconnect to another block if I remove the current connection by clicking the remove Icon.
Dragging the move icon to connect to a new block does not relocate connection and produces the console error below.

```
block.ts?6df9:108 
Uncaught (in promise) Error: Unable to swap destinationBlockId on null: {"first":{"blockId":"91ef29bf-c54d-400a-82dc-2f9e830c893d","exitId":"b5c86703-7c78-4976-9f72-b46c428353a4"},"second":null}
    at _callee4$ (block.ts?6df9:108)
    at tryCatch (runtime.js?96cf:45)
    at Generator.invoke [as _invoke] (runtime.js?96cf:274)
    at Generator.prototype.<computed> [as next] (runtime.js?96cf:97)
    at asyncGeneratorStep (asyncToGenerator.js?1da1:3)
    at _next (asyncToGenerator.js?1da1:25)
    at eval (asyncToGenerator.js?1da1:32)
    at new Promise (<anonymous>)
    at eval (asyncToGenerator.js?1da1:21)
    at Store.block_swapBlockExitDestinationBlockIds (block.ts?6df9:108)

```
```

**Comment by safydy-r** [comment]
Thanks @adeablay , we created a separated ticket for it https://viamoinc.atlassian.net/browse/VMO-3467

**Pull request merged**